### PR TITLE
OGM-1225 Generic test fixes

### PR DESCRIPTION
**Pull request by DavideD**
https://hibernate.atlassian.net/browse/OGM-1225

Fixes some simple issues with tests and add some comments

### CODE REVIEW

File: neo4j/src/test/java/org/hibernate/ogm/datastore/neo4j/test/mapping/UnidirectionalManyToManyTest.java [hunk 1]
```diff
@@ -22,57 +22,77 @@
  * @author Davide D'Alto
  */
 public class UnidirectionalManyToManyTest extends Neo4jJpaTestCase {
-	private AccountOwner owner;
-	private BankAccount soge;
-	private BankAccount barclays;
+
+	Student john;
```
Comment (gsmet) [inline neo4j/src/test/java/org/hibernate/ogm/datastore/neo4j/test/mapping/UnidirectionalManyToManyTest.java]: Let's keep them private.

File: core/src/test/java/org/hibernate/ogm/backendtck/associations/collection/manytomany/ManyToManyTest.java [hunk 2]
```diff
@@ -126,7 +127,8 @@ public void testManyToManyCompositeId() throws Exception {
 
 	private int expectedAssociationNumber() {
 		if ( EnumSet.of( NEO4J_EMBEDDED, NEO4J_REMOTE ).contains( TestHelper.getCurrentDialectType() ) ) {
-			// In Neo4j relationships are bidirectional
+			// Neo4j map an association as one relationship either if it's a bidirectional or unidirectional one.
+			// This means that when can only count considering each association as unidirectional.
```
Comment (gsmet) [inline core/src/test/java/org/hibernate/ogm/backendtck/associations/collection/manytomany/ManyToManyTest.java]: I think there's a missing word or something. I don't understand the sentence.

**Comment by DavideD** [comment]
@gsmet Fixed

**Comment by gsmet** [comment]
Fixed a few remaining typos (mostly in commit comments) and merged. Thanks.

**Pull request closed without merging**